src/main/java/org/jboss/remoting3/remote/RemoteConnectionProvider.java

```diff
@@ -204,7 +208,32 @@ public Cancellable cancel() {
         if (useSsl && destination instanceof InetSocketAddress) {
             if (xnioSsl == null) {
                 try {
-                    xnioSsl = xnio.getSslProvider(connectOptions);
+                    Boolean usePKCS = connectOptions.get(RemoteConnectionProviderFactory.JBOSS_AS_REMOTE_USEPKCS, false);
```

Comment:
I'm afraid that without model change (which we don't want) there is no other way how to pass data from client (except parameters as is used in this PR). This is only a slightly better way then using system properties as is suggested in workaround - which could have impact also on other processes running in JVM.